src/engine/strat_engine/engine.rs

```diff
@@ -79,10 +80,14 @@ impl StratEngine {
             }
         }
 
-        Ok(StratEngine {
-               pools: table,
-               watched_dev_last_event_nrs: HashMap::new(),
-           })
+        let engine = StratEngine {
+            pools: table,
+            watched_dev_last_event_nrs: HashMap::new(),
+        };
+
+        devlinks::setup_devlinks(engine.pools().into_iter())?;
```

Comment:
So long as ```Table::into_iter()``` doesn't consume all the pools this is good. It looks like it doesn't, but I'm not quite sure.

Reply:
(sorry for delay, fell off my radar!)

`Engine::pools()` returns a `Vec<&Pool>` by value, so this vec of trait objects is the thing being consumed rather than the pools themselves.

(`Engine::pools()` also calls into_iter(), but this is ok because Table::into_iter() acts like .iter() right now. See #678 or #509.)

Reply:
@mulkieran this ok with you?